frontend/src/data/Dialect.ts

```diff
@@ -32,14 +31,70 @@ export default class Dialect {
   }
 
   async fetchReport(baseURI: URI) {
-    const url = baseURI.clone().filename(this.path).suffix("json").href();
-    const response = await fetch(url);
+    const curVersionUrl = baseURI
+      .clone()
+      .filename(this.path)
+      .suffix("json")
+      .href();
+    const response = await fetch(curVersionUrl);
     const jsonl = await response.text();
-    const lines = jsonl
+    const curReportLines = jsonl
       .trim()
       .split(/\r?\n/)
       .map((line) => JSON.parse(line) as Record<string, unknown>);
-    return parseReportData(lines);
+
+    let curReport = parseReportData(curReportLines);
+
+    const prevVersionUrl = baseURI
+      .clone()
+      .directory("previous")
+      .filename(this.path)
+      .suffix("json")
+      .href();
+    const prevReportLines = await this.fetchReportMetadata(prevVersionUrl);
+    if (prevReportLines) {
+      const prevReportMetaData = JSON.parse(prevReportLines) as RunInfo;
+      curReport = this.compareReportToOld(curReport, prevReportMetaData);
+    }
+    return curReport;
+  }
+
+  async fetchReportMetadata(url: string) {
```

Comment:
This looks like what I'd expect, but also looks like the sort of thing I would expect to already *exist* somewhere -- I believe you if you looked and didn't see it, but let's still put it somewhere where we can replace it when/if we find something which does it.

So rather than doing it inline in a function called `fetchReportMetadata`, do it in some helper function which for any `fetch()` call streams the lines of the response (as a JS coroutine I think?). That likely exists somewhere, so at some point we can replace it when we find it, and until then you can use it to do this here.